### PR TITLE
Large files: preparation

### PR DESCRIPTION
**Pull request by bwbeach**
Start getting ready to implement large file support.  The first step will be to have complete unit tests for uploading and retrying.  This checkin gets things ready to write those unit tests.

### CODE REVIEW

File: b2/b2.py [hunk 1]
```diff
@@ -2320,8 +2450,8 @@ def sync(self, args):
                 print("+ %s" % filename)
                 if not os.path.exists(dirpath):
                     os.makedirs(dirpath)
-                url = self.api.get_download_url_for_fileid(remote_file['fileId'])
-                download_file_by_id_helper(self.api, url, filepath, authorization=True,)
+                download_dest = DownloadDestLocalFile(filepath, DoNothingProgressListener())
```
Comment (ppolewicz) [inline b2/b2.py]: Why no progress reporting?

**Comment by ppolewicz** [comment]
I reviewed all and I approve. Really good work there.

**Comment by bwbeach** [comment]
Thanks.  I'm trying not to conflict with your work on B2Session too much.

File: b2/b2.py (hunk 1 above)
Comment (bwbeach) [inline b2/b2.py]: Good question.  I'll take a look and see if it had progress reporting before.

**Pull request merged**